fluss-flink/fluss-flink-common/src/test/java/com/alibaba/fluss/flink/source/deserializer/FlussRowDeserializerTest.java

```diff
@@ -0,0 +1,154 @@
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+
+/**
+ * Test class for the {@link RowDataDeserializationSchema} that validates the conversion from Fluss
+ * {@link com.alibaba.fluss.record.LogRecord} to Flink's {@link RowData} format.
+ */
+public class FlussRowDeserializerTest {
+
+    private RowType rowType;
+    private RowDataDeserializationSchema schema;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        List<DataField> fields =
+                Arrays.asList(
+                        new DataField("orderId", DataTypes.BIGINT()),
+                        new DataField("itemId", DataTypes.BIGINT()),
+                        new DataField("amount", DataTypes.INT()),
+                        new DataField("address", DataTypes.STRING()));
+
+        rowType = new RowType(fields);
+        schema = getRowDataDeserializationSchema(rowType);
+    }
+
+    @Test
+    public void testDeserialize() throws Exception {
+        // Create test data
+        GenericRow row = new GenericRow(4);
+        row.setField(0, 100L);
+        row.setField(1, 10L);
+        row.setField(2, 45);
+        row.setField(3, BinaryString.fromString("Test addr"));
+
+        ScanRecord scanRecord = new ScanRecord(row);
+
+        RowDataDeserializationSchema deserializer = getRowDataDeserializationSchema(rowType);
+        RowData result = deserializer.deserialize(scanRecord);
+
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(row, scanRecord.getRow());
+    }
+
+    private @NotNull RowDataDeserializationSchema getRowDataDeserializationSchema(RowType rowType)
+            throws Exception {
+        RowDataDeserializationSchema deserializationSchema = new RowDataDeserializationSchema();
+        deserializationSchema.open(
+                new FlussDeserializationSchema.InitializationContext() {
+                    @Override
+                    public MetricGroup getMetricGroup() {
+                        return null;
+                    }
+
+                    @Override
+                    public UserCodeClassLoader getUserCodeClassLoader() {
+                        return null;
+                    }
+
+                    @Override
+                    public RowType getRowSchema() {
+                        return rowType;
+                    }
+                });
+        return deserializationSchema;
+    }
+
+    @Test
+    public void testDeserializeWithNullRecord() {
+        assertThrows(NullPointerException.class, () -> schema.deserialize(null));
+    }
+
+    @Test
+    public void testGetProducedType() {
+        TypeInformation<RowData> typeInfo = schema.getProducedType();
+
+        assertNotNull(typeInfo);
+        assertEquals(RowData.class, typeInfo.getTypeClass());
```

Comment:
assert the `typeInfo.createSerializer(new ExecutionConfig())` is an instance of `RowDataSerializer`